Source/DafnyCore/AST/Expressions/Expressions.cs

```diff
@@ -1928,21 +1929,25 @@ public enum AccumulationOperand { None, Left, Right }
     Contract.Invariant(E1 != null);
   }
 
-  public BinaryExpr(IToken tok, Opcode op, Expression e0, Expression e1)
+  public BinaryExpr(IToken tok, Opcode op, Expression e0, Expression e1, IToken prefixOp = null)
     : base(tok) {
     Contract.Requires(tok != null);
     Contract.Requires(e0 != null);
     Contract.Requires(e1 != null);
     this.Op = op;
     this.E0 = e0;
     this.E1 = e1;
+    this.PrefixOp = prefixOp;
+    if (prefixOp != null) {
```

Comment:
Can't you compute `prefixOp` using `e0.RangeToken.End.Next` ? Not having prefixOp as an extra parameter seems quite valuable.

Reply:
Ah, that may be possible. That's an artifact of doing most of this work before merging with PR #3070. Previously, `RangeToken` wasn't updated to include the leading `||` or `&&`, but now it is. I'll investigate.

Reply:
It is in fact possible to remove `prefixOp`, and it's much nicer that way. Thanks for pointing it out!